### ISSUE TITLE
Upcoming Features

### ISSUE THREAD

**Issue by Mole1803** [cfloutier/k2d2#3]
- [x]  Set the Periapsis and the Apoapsis with a feature
- [ ] function for changing Inclination 
- [ ] function for chanching the longitude of ascending node
- [ ] Manual Hohmann transfer
- [ ] Automatic Hohmann transfer

Feel free to add more features you would like to see in this Mod!

**Comment by cfloutier** [comment]
Excellent list 👍👍

**Comment by schlosrat** [comment]
Most of these are available in Flight Plan, which can now call K2-D2. I believe I can add a function for changing the LAN too if that's of value to you.
Also, Flight Plan has a library mode where you can call many of its methods for things like this. The entry points are all public methods in the base class, so these capabilities can be integrated either way.

**Comment by cfloutier** [comment]
This issue is about what mole wanted to make in the mod. as he does have time to continue for the moment, I let the issue opened. 
Those feature can stay in dev mode only. as it is a kind of laboratory for @Mole1803. 

from my side I prefer to let FlightPlan do the stuff :)

**Comment by solarisfire** [comment]
Can the "lift" page be improved to give a more settings to change a launch ascent?

![image](https://github.com/cfloutier/k2d2/assets/420952/5c682ac2-6805-43a7-8bda-a597393e504c)

**Comment by cfloutier** [comment]
Of course it's in the future plan.
I won't fit to mechjeb feature the code is fully different

Le jeu. 11 mai 2023 à 21:44, Stephen Greenham ***@***.***> a
écrit :

**Comment by schlosrat** [comment]
I'd like to see a control for forcing the roll of the vessel to a particular value. Yaw and pitch will be whatever they need to be, but if I've got only two side boosters then I prefer to roll the vessel to a chosen angle so that when the boosters separate they are less likely to collide with the remaining central core. 

I don't think we need a complex set of ascent controls though. At least not right now. At some point when heating is added to the game we might, but that's uncertain. Personally, I like the simple clean interface you've got now where I just need to set the final altitude. I've found it to work well with all the rockets I've launched with it.

It would be nice to also be able to launch into the plane of a target's orbit. I'm hopeful that FP can help supply the info K2-D2 would need for that.

**Comment by cfloutier** [comment]
Yes @schlosrat. I need it too and it could be an option available anytime even if no pilot is active. I'll try to do it soon as I've not really dig in this way

**Comment by GabeADemarco** [comment]
**Hi, thanks for the mod! I love it!**

Suggestion for Lift module:
How about instead of having a Max Throttle setting, or not instead but at the same time... have a setting for desired TWR regardless of rocket capabilities?
For instance, let's say i have a rocket with TWR = 3, for lift off i believe some deltaV would be lost due to aerodynamic pressure, i'm not 100% on this but as far as I know, that's how it works. That setting could be for example @1.5, and K2-D2 would manage the throttle to achieve that TWR.

Thanks!

**Comment by schlosrat** [comment]
@GabeADemarco I would suggest following the approach of MechJeb for this. They've already worked out a good approach that considers these things such as allowing to limit throttle (also keep limited throttle over xxx), limit acceleration, limit Q (maximum dynamic pressure), limit Angle of Attack (AoA), Dynamic pressure at which to fade out the limitation of AoA.
Borrowing an approach from MJ does not necessarily mean borrowing code - just taking advantage of the list of features they've found worth implementing and keeping around.

**Comment by GabeADemarco** [comment]
@schlosrat yep that's what i meant (exactly), and also what i'm used to use, but i didn't want to impose a certain way because for once i have no idea what i'm talking about, and also, because modders can do whatever they want with their mods, so i thought it would be more "respectful" to just express the idea instead of "do mechjeb".
Thanks!!

**Comment by cfloutier** [comment]
@GabeADemarco TWR that's an excellent idea. and more it would be quite easy to code cause I need the a precise TWR during the Land and Drone Pilots. So I know it would be easy to get. I'll try to code it and show current TWR on the K2D2 screen

**Comment by cfloutier** [comment]
@schlosrat thanks for defending my approach of pure naive new way to code it. 
I does not want to reproduce mechjeb. But some ideas there are excellents and it would be stupid to not take them too because they were in mechjeb.

**Comment by bsiebens** [comment]
Some sort of landing capability where you could specify the touchdown coordinates would be nice to help with more precision landing)

**Comment by cfloutier** [comment]
Yes i' would be so cool. This is in my todo list with schlorat

Le lun. 25 mars 2024, 08:59, Bernard Siebens ***@***.***> a
écrit :
